### PR TITLE
Fixed a prettyprinter and boolean confusion bug in coverage

### LINKED ISSUE #4239: Boolean coverage confusion
Currently the boolean coverage annotations that are put in by the PlutusTx compiler confuse the `True` and `False` cases. Furthermore, the metadata annotations are not pretty-printed correctly (not one per line indented). This is fixed in #4238.

### PR DESCRIPTION
**Pull request by MaximilianAlgehed**
Pre-submit checklist:
- Branch
    - [X] Tests are provided (if possible)
    - [X] Commit sequence broadly makes sense
    - [X] Key commits have useful messages
    - [X] Relevant tickets are mentioned in commit messages
    - [X] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [X] Self-reviewed the diff
    - [X] Useful pull request description
    - [X] Reviewer requested
This fixes #4239

### CODE REVIEW

File: plutus-tx-plugin/test/Budget/find.budget.golden [hunk 1]
```diff
@@ -1 +1 @@
-ExBudget {exBudgetCPU = ExCPU 49371287, exBudgetMemory = ExMemory 161020}
\ No newline at end of file
+ExBudget {exBudgetCPU = ExCPU 49758336, exBudgetMemory = ExMemory 162320}
```
Comment (michaelpj) [inline plutus-tx-plugin/test/Budget/find.budget.golden:1]: I suspect these test outputs are flip-flopping, you might just want to revert them.

File: plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs [hunk 2]
```diff
@@ -843,8 +843,8 @@ coverageCompile originalExpr exprType src compiledTerm covT =
             let mkMetadata = CoverageMetadata . foldMap (Set.singleton . ApplicationHeadSymbol . GHC.getOccString)
             fc <- addBoolCaseToCoverageIndex (toCovLoc src) False (mkMetadata headSymName)
             tc <- addBoolCaseToCoverageIndex (toCovLoc src) True (mkMetadata headSymName)
-            pure $ PLC.mkIterApp () traceBoolCompiled [PLC.mkConstant () (T.pack . show $ fc), PLC.mkConstant () (T.pack . show $ tc), compiledTerm]
-          _ -> throwSd CompilationError $ "Lookup of traceBool failed. Expected to get AnId but saw: " GHC.<+> (GHC.ppr traceBoolThing)
+            pure $ PLC.mkIterApp () traceBoolCompiled [PLC.mkConstant () (T.pack . show $ tc), PLC.mkConstant () (T.pack . show $ fc), compiledTerm]
+          _ -> throwSd CompilationError $ "Lookup of traceBool failed. Expected to get An Id but saw: " GHC.<+> (GHC.ppr traceBoolThing)
```
Comment (michaelpj) [inline plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs:847]: previous version seemed more correct

**Review by michaelpj — APPROVED** [review]
:facepalm:

File: plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs (hunk 2 above)
Comment (MaximilianAlgehed) [reply plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs:847]: You're right. Too many "fixes" at the same time.

**Pull request merged**